tests/core/full_node/test_tx_processing_queue.py

```diff
@@ -0,0 +1,212 @@
+    spend_bundle = SpendBundle(coin_spends, blspy.G2Element())
+    return spend_bundle
+
+
+# we only want to call this once as it can take minutes if we call it many times.
+standard_spend_bundle = make_standard_spend_bundle(3)
+
+
+def get_peer_id() -> bytes32:
+    return bytes32(token_bytes(32))
+
+
+def get_transaction_queue_entry(peer_id: bytes32) -> TransactionQueueEntry:
+    sb: SpendBundle = standard_spend_bundle
+    return TransactionQueueEntry(
+        sb,
+        peer_id,  # we cheat a bit here by reusing transaction_bytes to store our peer_id
+        sb.name(),
+        None,
+        False,
+    )
+
+
+class TestTransactionQueue:
+    @pytest.fixture(scope="function")
+    def transaction_queue(self) -> TransactionQueue:
+        return TransactionQueue(1000, log)
+
+    @pytest.mark.asyncio
+    async def test_local_txs(self, transaction_queue: TransactionQueue) -> None:
+        # test 1 tx
+        first_tx = get_transaction_queue_entry(get_peer_id())
+        await transaction_queue.put(first_tx, None)
+
+        assert transaction_queue._index_to_peer_map == []
+        assert transaction_queue._queue_length._value == 1
+
+        result1 = await transaction_queue.pop()
+
+        assert transaction_queue._queue_length._value == 0
+        assert result1 == first_tx
+
+        # test 2000 txs
+        num_txs = 2000
+        list_txs = [get_transaction_queue_entry(get_peer_id()) for _ in range(num_txs)]
+        for tx in list_txs:
+            await transaction_queue.put(tx, None)
+
+        assert transaction_queue._queue_length._value == num_txs  # check that all are included
+        assert transaction_queue._index_to_peer_map == []  # sanity checking
+
+        resulting_txs = []
+        for _ in range(num_txs):
+            resulting_txs.append(await transaction_queue.pop())
+
+        assert transaction_queue._queue_length._value == 0  # check that all are removed
+        for i in range(num_txs):
+            assert list_txs[i] == resulting_txs[i]
+
+    @pytest.mark.asyncio
+    async def test_one_peer_and_await(self, transaction_queue: TransactionQueue) -> None:
+        num_txs = 100
+        peer_id = bytes32(token_bytes(32))
+
+        list_txs = [get_transaction_queue_entry(peer_id) for _ in range(num_txs)]
+        for tx in list_txs:
+            await transaction_queue.put(tx, peer_id)
+
+        assert transaction_queue._queue_length._value == num_txs  # check that all are included
+        assert transaction_queue._index_to_peer_map == [peer_id]  # sanity checking
+
+        resulting_txs = []
+        for _ in range(num_txs):
+            resulting_txs.append(await transaction_queue.pop())
+
+        assert transaction_queue._queue_length._value == 0  # check that all are removed
+        for i in range(num_txs):
+            assert list_txs[i] == resulting_txs[i]
+
+        with pytest.raises(asyncio.TimeoutError):
```

Comment:
is this actually waiting in wall-clock time? If so, I think we should test this without sleeping. e.g. we could create a task and ensure it's blocked in the `pop()`